tfjs-tflite-node-codelab/webnn_inference_working/renderer.js

```diff
@@ -0,0 +1,156 @@
+const tf = require('@tensorflow/tfjs');
+// CODELAB part 2: Import the delegate here.
+const {WebNNDelegate} = require('webnn-tflite-delegate');
+const fs = require('fs');
+const Stats = require('stats.js');
+
+async function getWebcam() {
+  const webcam = document.createElement('video');
+  webcam.width = 224;
+  webcam.height = 224;
+  const stream = await navigator.mediaDevices.getUserMedia({
+    audio: false,
+    video: {
+        mandatory: {
+          minHeight: webcam.height,
+          minWidth: webcam.width,
+          maxHeight: webcam.height,
+          maxWidth: webcam.width,
+        }
+    }
+  });
+  // Flip the image so the video looks like a mirror.
+  webcam.style.transform = 'scaleX(-1)';
+  webcam.srcObject = stream;
+  webcam.play();
+  return webcam;
+}
+
+async function main() {
+  const stats = new Stats();
+  document.body.appendChild(stats.dom);
+
+  const webcam = await getWebcam();
+  document.body.appendChild(webcam);
+
+  // For displaying results
+  const resultHeader = document.createElement('h1');
+  resultHeader.innerHTML = 'Classification: ';
+  document.body.appendChild(resultHeader);
+
+  const resultList = document.createElement('ul');
+  document.body.appendChild(resultList);
+
+  function showPrediction(data, labels) {
+    resultList.innerHTML = ''; // Clear the list
+    for (let i = 0; i < data.length; i++) {
+      const percent = data[i].toFixed(1);
+      const resultText = `${labels[i]}: ${percent}`;
+      const result = document.createElement('li');
+      result.innerHTML = resultText;
+      resultList.appendChild(result);
+    }
+  }
+
+  // CODELAB part 1: Load the model here.
+  const modelPath = './model/model_unquant.tflite';
+  const cpuModel = await loadTFLiteModel(modelPath);
+  const labels = fs.readFileSync('./model/labels.txt', 'utf8')
+        .split(/\r?\n/);
+
+  // CODELAB part 2: Load the delegate model here.
+  let webnnModel = await loadTFLiteModel(modelPath, {
+    delegates: [new WebNNDelegate([['webnn_device', '0']])],
+  });
+
+  // CODELAB part 1: Set up tf.data.webcam here.
+  const tensorCam = await tf.data.webcam(webcam);
+
+  // CODELAB part 2: Create the delegate button here.
+  let useWebNNDelegate = false;
+
+  // Create elements for WebNN device selection
+  const divElem = document.createElement('div');
+  divElem.innerHTML = '<br/>WebNN Device: ';
+  const selectElem = document.createElement('select');
+  divElem.appendChild(selectElem);
+
+  const webnnDevices = ['Default', 'GPU', 'CPU'];
+  // append the options
+  for (let i = 0; i < webnnDevices.length; i++) {
+    var optionElem = document.createElement('option');
+    optionElem.value = i;
+    optionElem.text = webnnDevices[i];
+    selectElem.appendChild(optionElem);
+  }
+  selectElem.addEventListener('change', async () => {
+    webnnModel = await loadTFLiteModel(modelPath, {
+      delegates: [new WebNNDelegate([['webnn_device', selectElem.value]])],
```

Comment:
This also needs to be updated due to #26.

Reply:
Thanks, done!